src/main/java/com/purbon/kafka/topology/BuilderCLI.java

```diff
@@ -36,6 +40,12 @@
   public static final String VERSION_OPTION = "version";
   public static final String VERSION_DESC = "Prints useful version information.";
 
+  public static final String EXTRACT_OPTION = "extract";
+  public static final String EXTRACT_DESC = "Extract to current cluster state and outputs as JSON";
+
+  public static final String RECOVER_OPTION = "recover";
+  public static final String RECOVER_DESC = "Recovers previously extracted state";
```

Comment:
is recover, something like applying the original topology files? what is the difference in your mind?

Reply:
Yes, but it applies a cluster-state (FlatDescription). The reason for this is two-fold: first of all, a cluster whose topics were not created using the KTB might not conform to KTB's naming convention. Hence we extract just a flat description (just topic names and configs as wells as ACLs, and later RBAC rolebindings (which aren't currently implemented)) which does not store any hierarchical information.